community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/pipes/ExecuteUpdateCommandsPipe.scala

```diff
@@ -69,10 +69,7 @@ case class ExecuteUpdateCommandsPipe(source: Pipe, commands: Seq[UpdateAction])(
 
   def sourceSymbols: SymbolTable = source.symbols
 
-  override def localEffects = source match {
-    case pipe: ExecuteUpdateCommandsPipe => pipe.source.localEffects
-    case _ => commands.effects(source.symbols)
-  }
+  override def localEffects = commands.effects(sourceSymbols)
```

Comment:
I remember fixing this once, there were failures when you had a `ExecuteCommandsPipe` on top of a `ExecuteCommandsPipe`, in that case it was wrong to just look at the commands. Have these refactorings removed that problem?